posts/the-radon-nikodym-theorem-in-lean.md

```diff
@@ -0,0 +1,162 @@
+---
+author: 'Kexing Ying'
+category: 'New in mathlib'
+date: 2021-09-22 12:51:14 UTC+02:00
+description: ''
+has_math: true
+link: ''
+slug: the-radon-nikodym-theorem-in-lean
+tags: ''
+title: The Radon-Nikodym theorem in Lean
+type: text
+---
+
+I have for the past two months been working on formalising the Radon-Nikodym theorem 
+in Lean, and with [PR #9065](https://github.com/leanprover-community/mathlib/pull/9065) 
+merged into mathlib, this journey seems to have finally come to an end. 
+
+The Radon-Nikodym theorem provides a necessary and sufficient condition for 
+comparing two measures, and allows us (under certain conditions) to express 
+one measure in terms of another.
+The Radon-Nikodym theorem is an important result in measure theory and has a
+wide range of applications in different fields of mathematics. Most notably, 
+it can be applied in probability theory in the definition 
+of the conditional expectation and in mathematical finance through the Girsanov 
+theorem[^1].
+
+[^1]: [wikipedia.org/wiki/Radon-Nikodym_theorem#Applications](https://en.wikipedia.org/wiki/Radon%E2%80%93Nikodym_theorem#Applications)
+
+Given two measures $\mu$ and $\nu$ on $\alpha$, we say that $\mu$ is 
+*absolutely continuous* with respect to $\nu$ (and write $\mu \ll \nu$) if for all 
+$S \subseteq \alpha$, $\nu(S) = 0$ implies $\mu(S) = 0$. Absolutely continuity 
+is an important notion for measures and we would like to establish a condition 
+for when it is true. 
+
+Given a measure $\mu$ on $\alpha$ and a measurable function 
+$f : \alpha \to \overline{\mathbb{R}}_{\ge 0}$, the set function 
+$$S \mapsto \int_S f \text{ d} \mu$$
+is also a measure on $\alpha$ and we denote this measure by $f\mu$. It is 
+easy, and intuitive to see that $f\mu \ll \mu$, however, it is not clear 
+whether the reverse is true. The Radon-Nikodym theorem proves the reverse 
+implication for certain measures. In particular, the classical Radon-Nikodym 
+theorem states that, two $\sigma$-finite measures $\mu, \nu$ satisfies $\mu \ll \nu$ 
+if and only if there exists a *essentially unique* measurable function $f$ 
+such that $\mu = f\nu$. This function is known as the Radon-Nikodym derivative 
+and is denoted by $\frac{\text{d}\mu}{\text{d}\nu}$.
+
+In Lean, this is shown by 
+```lean
+theorem absolutely_continuous_iff_with_density_radon_nikodym_deriv_eq
+  {μ ν : measure α} [have_lebesgue_decomposition μ ν] :
+  μ ≪ ν ↔ ν.with_density (radon_nikodym_deriv μ ν) = μ
+```
```

Comment:
This is very hard to related to the above discussion because of the `have_lebesgue_decomposition μ ν` assumption which doesn't appear at all above.

Reply:
I've moved the lean code after the section about `have_lebesgue_decomposition` so perhaps it makes more sense